src/components/Board/Board.js

```diff
@@ -1,42 +1,38 @@
 import "./Board.scss";
 import Square from "../Square/Square";
+import Turns from "../Turns/Turns";
 import reducer, { initialStates } from "../../reducers";
 import React from "react";
-import redBullet from "../../assets/images/red-bullet.png";
-import blueBullet from "../../assets/images/blue-bullet.png";
-import classNames from "classnames";
 import Confetti from "react-confetti";
 import data from "../../assets/data.json";
-import { createRandomNumber } from "../../helper/util";
+import { createRandomNumber, createSpiralArray } from "../../helper/util";
 import Die from "../Die/Die";
 
 // This function searches through the snakes list to check if it contains a specific square number, if so, it returns the information of that snake.
 const foundSnake = (squareNumber) => {
   const snakes = data.snakes;
+  let result = undefined;
 
   // I would re-write it in a functional way, using array functions :)
-  let result = undefined;
-  for (let i = 0; i < snakes.length; i++) {
-    if (snakes[i].source === squareNumber) {
-      result = snakes[i];
-      break;
-    }
-  }
+  //******************************************* QUESTION ********************************************
+  //I'm not sure what you meant by "array functions", but if you mean using .fill().map(), I didn't use it.
+  //I've heard that as the map() function creates another array with the same size, we ends up having 2 arrays in the memory which
+  //is not a good idea for the purpose of optimization. Please let me know about 1.what you meant and 2.if I'm wrong. Thanks:)
+
+  snakes.forEach((element) => {
+    if (element.source === squareNumber) result = element;
+  });
   return result;
```

Comment:
Sorry, my comment wasn't too precise.
You can use `.find()` or `.findIndex()` on your array.
`.find()` will return you the actual value based on a predicate. It stops the iteration once the predicate matches. If the value doesn't exist, it returns `undefined`

`.findIndex()` will return you the index of the value based on a predicate. It stops the iteration once the predicate matches.  If the value doesn't exist, it returns `-1`

eg.:
```js
const people = [
  { name: 'Jose', age: 16 },
  { name: 'Constantine', age: 18 },
  { name: 'Kitty', age: 17 },
]

people.findIndex(p => p.name === 'Jose') // ---> 0
people.findIndex(p => p.name === 'nobody') // ---> -1

people.find(p => p.name === 'Jose') // ---> { name: 'Jose', age: 16 },
people.find(p => p.name === 'nobody') // ---> undefined
```

Reply:
Now how easy it works! Perfect